### ISSUE TITLE
Cannot pass `EcsOptimizedImage` to `IMachine` parameter in construct

### ISSUE THREAD

**Issue by Chriscbr** [cdklabs/decdk#50]
Example:

```
"$schema": "./cdk.schema.json"
Resources:
  MyVpc:
    Type: aws-cdk-lib.aws_ec2.Vpc
    Properties:
      maxAzs: 2
  MyFleet:
    Type: aws-cdk-lib.aws_autoscaling.AutoScalingGroup
    Properties:
      vpc:
        Ref: VPC
      instanceType:
        of:
          instanceClass: T2
          instanceSize: XLARGE
      machineImage:
        aws-cdk-lib.aws_ecs.EcsOptimizedImage:
          amazonLinux2: {}
      desiredCapacity: 3,
```

which is equivalent to this TypeScript CDK app:

```ts
    const vpc = new ec2.Vpc(this, 'MyVpc', { maxAzs: 2 });

    const asg = new autoscaling.AutoScalingGroup(this, 'MyFleet', {
      instanceType: ec2.InstanceType.of(ec2.InstanceClass.T2, ec2.InstanceSize.XLARGE),
      machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
      desiredCapacity: 3,
      vpc,
    });
```

...produces the error:

```
Error: Cannot find schema for aws-cdk-lib.aws_ecs.EcsOptimizedImage
```

**Comment by mrgrain** [comment]
Related to #201

**Comment by mrgrain** [comment]
Static methods are defined as a FQN. So the template should be like this:
```
      machineImage:
        aws-cdk-lib.aws_ecs.EcsOptimizedImage.amazonLinux2: {}
```

**Comment by mrgrain** [comment]
Current json schema wants me to write:
```
      machineImage:
        amazonLinux2: {}
```

However this cannot be resolved unambiguously as  the static method could exist on multiple factory classes, not just `EcsOptimizedImage`.

tl;dr this issue is strongly related to #241 and blocked by it.
